packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import { Getter, Action, Plugin, Getters } from '@devexpress/dx-react-core';
+import {
+  recalculateBounds, calculateRequestedRange, virtualRowsWithCache,
+  trimRowsToInterval, intervalUtil, emptyVirtualRows, plainRows, loadedRowsStart,
+} from '@devexpress/dx-grid-core';
+import { VirtualTableStateProps, VirtualTableStateState } from '../../types';
+
+const virtualRowsComputed = (
+  { start, rows, virtualRowsCache }: Getters,
+) => virtualRowsWithCache(start, rows, virtualRowsCache);
+
+const rowsComputed = ({ virtualRows }: Getters) => plainRows(virtualRows);
+
+const loadedRowsStartComputed = ({ virtualRows }: Getters) => loadedRowsStart(virtualRows);
+
+// tslint:disable-next-line: max-line-length
+class VirtualTableStateBase extends React.PureComponent<VirtualTableStateProps, VirtualTableStateState> {
+  static defaultProps = {
+    virtualPageSize: 100,
+  };
+  requestTimer: any = 0;
```

Comment:
Why is it not `number`?

Reply:
Previously setTimeout resolved to NodeJS.Timeout type, so now window.setTimeout used instead